staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go

```diff
@@ -117,11 +117,39 @@ func decodeVersionedSet(encodedVersionedSet *metav1.ManagedFieldsEntry) (version
 	if encodedVersionedSet.Fields != nil {
 		fields = *encodedVersionedSet.Fields
 	}
-	set, err := FieldsToSet(fields)
-	if err != nil {
-		return nil, fmt.Errorf("error decoding set: %v", err)
+	return &cachedVersionedSet{
+		encodedSet: fields,
+		apiVersion: fieldpath.APIVersion(encodedVersionedSet.APIVersion),
+		applied:    encodedVersionedSet.Operation == metav1.ManagedFieldsOperationApply,
+	}, nil
+}
+
+type cachedVersionedSet struct {
+	encodedSet metav1.Fields
+	set        *fieldpath.Set
+	apiVersion fieldpath.APIVersion
+	applied    bool
+}
+
+var _ fieldpath.VersionedSet = &cachedVersionedSet{}
+
+func (c *cachedVersionedSet) Set() *fieldpath.Set {
+	if c.set == nil {
+		set, err := FieldsToSet(c.encodedSet)
+		if err == nil {
```

Comment:
That is really unfortunate. This means that if someone changes a managedField in an update, it's not going to fail and just return an empty set, is that what we want?

Reply:
No probably not. We could collect any errors here and return them at the end?

Reply:
Actually, why not just add an error return value to Set() in the VersionedSet interface. Should be simpler.